Vwitter/View Controllers/CreateGroupViewController.m

```diff
@@ -91,57 +91,34 @@ - (void)tableView:(UITableView *)tableView didDeselectRowAtIndexPath:(NSIndexPat
 
 }
 
-//refactor to be parse cloud function
 - (IBAction)didCreateGroup:(id)sender {
     
     [MBProgressHUD showHUDAddedTo:self.view animated:YES];
     
-    GroupDetails *currentGroup = [[GroupDetails alloc] initWithGroupName:self.groupNameField.text withGroupAuthor:[VWUser currentUser]];
+    NSMutableArray *groupMemberships = [NSMutableArray arrayWithArray:[self.arrayOfSelectedUserAudience allObjects]];
     
-    NSMutableArray *groupMemberships = [[NSMutableArray alloc] init];
+    NSMutableArray *groupMembershipIds = [groupMemberships valueForKey:@"objectId"];
     
     __weak typeof(self) weakSelf = self;
-    [currentGroup saveInBackgroundWithBlock: ^(BOOL succeeded, NSError * _Nullable error) {
+    [PFCloud callFunctionInBackground:@"createGroup"
+                       withParameters:@{@"authorId":[VWUser currentUser].objectId, @"groupName":self.groupNameField.text, @"groupMembershipIds":groupMembershipIds}
+                                block:^(id groups, NSError *error) {
```

Comment:
nice!!! This is so much simpler! 

Still, this is also very dangerous. You should do input validation to make sure `self.groupNameField.text` for example is not nil/empty string (and use your standard UIViewController+ErrorAlert helper to show a helpful error to the user if it is), same with validating the other fields like authorId